doc/website/blog/2022-10-10-json-append.mdx

```diff
@@ -0,0 +1,296 @@
+	Body string `json:"body"`
+}
+
+func (User) BlogPost() []ent.Field {
+	field.JSON("comments", []Comment{})
+}
+```
+
+Ent provides a convenient API for reading and writing values to JSON columns, as well
+as applying predicates (using [`sqljson`](https://entgo.io/docs/predicates/#json-predicates)):
+```go
+func TestEntJSON(t *testing.T) {
+	client := enttest.Open(t, "sqlite3", "file:ent?mode=memory&cache=shared&_fk=1")
+	ctx := context.Background()
+	// Insert a user with two comments.
+	client.User.Create().
+		SetComments([]schema.Comment{
+			{Title: "hello"},
+			{Title: "goodbye"},
+		}).
+		SaveX(ctx)
+
+	// Count how many users have more than one comment.
+	count := client.User.Query().
+		Where(func(s *sql.Selector) {
+			s.Where(
+				sqljson.LenGT(user.FieldComments, 0, sqljson.DotPath(".")),
+			)
+		}).
+		CountX(ctx)
+	fmt.Printf("count: %d", count)
+	// Prints: count: 1
+}
+```
+
+### Appending values to fields in JSON columns
+
+In many cases, it is useful to append a value to a list in a JSON column.
+Preferably, appends are implemented in a way that is _atomic_, that is, without
+needing to read the current value and writing the entire new value. The reason
+for this, is that if two calls try to append the value concurrently, both will
+read the same _current_ value from the database, and write their own updated version
+roughly at the same time. Unless [optimistic locking](2021-07-22-database-locking-techniques-with-ent.md)
+is used, both writes will succeed, but the final result will only include one of
+the new values. 
+
+To overcome this race condition, we can let the database take care of the synchronization
+between both calls by using a clever UPDATE query. The intuition for this solution
+is similar to how counters are incremented in many applications. Instead of running:
+```sql
+SELECT points from leaderboard where user='rotemtam' 
+```
+Reading the result (lets say its 1000), incrementing the value  in process (1000+1=1001) and writing the new sum
+verbatim:
+```sql
+UPDATE leaderboard SET points=1001 where user='rotemtam'
+```
+Developers can use a query such as:
+```sql
+UPDATE leaderboard SET points=points+1 where user='rotemtam'
+```
+
+In a similar way, to avoid the need to synchronize writes using optimistic locking
+or some other mechanism, let's see how we can leverage the database's capability to
+perform them concurrently in a safe manner.
+
+Two things to note as we are constructing this query. First, the syntax for working
+with JSON values varies a bit between different database vendors, as you will see in
+the examples below. Second, a query for appending a value to a list in a JSON document
+needs to handle at least two edge cases:
+1. The field we want to append to doesn't exist yet in the JSON document.
+2. The field exists but is set to JSON `null`.
+
+Here is what such a query might look like for appending a value `new_val` to a field named `a`
+in a column `c` for table `t` in different dialects:
```

Comment:
I think this gets more readable if we don't use this "mathematical" approach, but a real life example.